examples/ner/run_ner.py

```diff
@@ -596,26 +600,69 @@ def main():
 
     args.model_type = args.model_type.lower()
     config_class, model_class, tokenizer_class = MODEL_CLASSES[args.model_type]
-    config = config_class.from_pretrained(
-        args.config_name if args.config_name else args.model_name_or_path,
-        num_labels=num_labels,
-        id2label={str(i): label for i, label in enumerate(labels)},
-        label2id={label: i for i, label in enumerate(labels)},
-        cache_dir=args.cache_dir if args.cache_dir else None,
-    )
     tokenizer_args = {k: v for k, v in vars(args).items() if v is not None and k in TOKENIZER_ARGS}
     logger.info("Tokenizer arguments: %s", tokenizer_args)
     tokenizer = tokenizer_class.from_pretrained(
         args.tokenizer_name if args.tokenizer_name else args.model_name_or_path,
         cache_dir=args.cache_dir if args.cache_dir else None,
         **tokenizer_args,
     )
-    model = model_class.from_pretrained(
-        args.model_name_or_path,
-        from_tf=bool(".ckpt" in args.model_name_or_path),
-        config=config,
-        cache_dir=args.cache_dir if args.cache_dir else None,
-    )
+
+    if args.has_new_labels and os.path.exists(args.model_name_or_path):
+        old_config = config_class.from_pretrained(
+            args.config_name if args.config_name else args.model_name_or_path,
+            cache_dir=args.cache_dir if args.cache_dir else None,
+        )
+        old_model = model_class.from_pretrained(
+            args.model_name_or_path,
+            from_tf=bool(".ckpt" in args.model_name_or_path),
+            config=old_config,
+            cache_dir=args.cache_dir if args.cache_dir else None,
+        )
+        old_num_labels = old_config.num_labels
+        new_labels = [label for label in labels if label not in old_config.label2id.keys()]
+        id2label = {**old_config.id2label, **{i+old_num_labels: label for i, label in enumerate(new_labels)}}
+        label2id = {**old_config.label2id, **{label: i+old_num_labels for i, label in enumerate(new_labels)}}
```

Comment:
You could also add code to remove old classifications if you wanted? So you would ignore some of the old labels if you wanted to. But then you'd have to remove/ignore the correct nodes in the data copying step.

It's probably easier to just remove the whole classification layer at that point?

Reply:
It's unfortunately so much worse than just copying in part of this. The classifier label <-> ids mappings are not used for ordering the weights matrix and bias vector - it re-alphabetizes the labels before it does this! I'm leaning on the fact that dictionaries retain order in python3.7+.

Reply:
Ahhhh!! That's terrible. Maybe leave that for another time unless you feel you're close. Ah yes, python3.7+ dicts are basically `OrderedDicts` now.

Reply:
Update for clarity and posterity:  my current best guess is that this depends on the ordering of the labels.txt file made through preprocessing. It's working reliably with the revisions which are presently in place.